SerialPrograms/Source/PokemonSV/Inference/Overworld/PokemonSV_DirectionDetector.cpp

```diff
@@ -0,0 +1,169 @@
+/*  Direction Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/GlobalSettingsPanel.h"
+#include "Common/Cpp/Containers/FixedLimitVector.tpp"
+#include "CommonFramework/ImageMatch/WaterfillTemplateMatcher.h"
+#include "CommonFramework/ImageTypes/ImageViewRGB32.h"
+#include "CommonFramework/ImageTools/WaterfillUtilities.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+#include "CommonFramework/ImageTools/ImageManip.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "CommonFramework/Tools/DebugDumper.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "Kernels/Waterfill/Kernels_Waterfill_Types.h"
+#include "PokemonSV_DirectionDetector.h"
+#include <cmath>
+#include <iostream>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+//using std::cout;
+//using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+
+class DirectionMatcher : public ImageMatch::WaterfillTemplateMatcher{
+public:
+    
+    DirectionMatcher() : WaterfillTemplateMatcher(
+        "PokemonSV/Map/DirectionIcon-Template.png", Color(0, 100, 30), Color(50, 230, 85), 50
```

Comment:
Don't forget to PR this file.

Reply:
https://github.com/PokemonAutomation/Packages/pull/29